packages/shared/src/lib/core/network/stores/networks.store.ts

```diff
@@ -1,32 +1,43 @@
-import { Readable, derived, get } from 'svelte/store'
+import { Writable, writable, get } from 'svelte/store'
 
 import { activeProfile } from '@core/profile/stores'
 
 import { IscpChain, StardustNetwork } from '../classes'
-import { IEvmNetwork, IStardustNetwork } from '../interfaces'
-import { Network, NetworkId } from '../types'
+import { IEvmNetwork, IIscpEvmNetworkConfiguration, IStardustNetwork } from '../interfaces'
+import { EvmNetworkConfiguration, Network, NetworkId } from '../types'
 import { EvmNetworkType, NetworkNamespace } from '../enums'
 
-export const networks: Readable<Network[] | undefined> = derived([activeProfile], ([$activeProfile]) => {
-    if ($activeProfile && $activeProfile.network) {
-        const stardustNetwork = new StardustNetwork($activeProfile.network)
-        const chains = $activeProfile.network.chainConfigurations
+export const networks: Writable<Network[]> = writable([])
+
+export function initializeNetworks(): void {
+    const profile = get(activeProfile)
+    if (profile && profile.network) {
+        const stardustNetwork = new StardustNetwork(profile.network)
+        const chains = profile.network.chainConfigurations
             .map((chainConfiguration) => {
                 switch (chainConfiguration.type) {
                     case EvmNetworkType.Iscp:
                         return new IscpChain(chainConfiguration)
-                    case EvmNetworkType.PureEvm:
-                        return undefined
                     default:
                         return undefined
                 }
             })
             .filter(Boolean) as IEvmNetwork[]
-        return [stardustNetwork, ...chains]
-    } else {
-        return undefined
+        networks.set([stardustNetwork, ...chains])
     }
-})
+}
+
+export function addNetwork(chainConfiguration: EvmNetworkConfiguration): void {
+    const network = getNetwork(chainConfiguration.id)
+    if (network) {
+        return
+    }
+
+    networks.update((networks) => {
+        networks.push(new IscpChain(chainConfiguration as IIscpEvmNetworkConfiguration))
```

Comment:
first check if the network isnt yet in this list

Reply:
That's what happens in l.31-33. Are you missing some specific check?